drivers/tee/optee/rpmb.c

```diff
@@ -131,8 +141,12 @@ static u32 rpmb_process_request(struct optee_private *priv, void *req,
 		mmc = get_mmc(priv, sreq->dev_id);
 		if (!mmc)
 			return TEE_ERROR_ITEM_NOT_FOUND;
-		if (mmc_rpmb_route_frames(mmc, RPMB_REQ_DATA(req),
-					  req_size - sizeof(struct rpmb_req),
+
+		if (!rpmb_frame_buffer)
```

Comment:
i would remove the change at line 72 and add here:
```
			if (!rpmb_frame_buffer) {
				rpmb_frame_buffer = malloc(RPMB_FRAME_BUFFER_SIZE);
				if (!rpmb_frame_buffer)
					return TEE_ERROR_OUT_OF_MEMORY;
			}
			memcpy(rpmb_frame_buffer, RPMB_REQ_DATA(req), RPMB_FRAME_SIZE);
```

Also, here we're not sure req_size == RPMB_FRAME_BUFFER_SIZE + sizeof(sreq);
If it differs from, then we must allocation/release the RPMB temporary buffer at each call :(

Reply:
I would prefer a persistence of the buffer, this way a new buffer will be allocated only if it needs more space. In any case the buffer is bonded by the request size so wont be bothered by previous data, and can be checked for free at release of mmc.
Something like this for exemple:
``` C
static size_t rpmb_frame_buffer_sz = 512;
static void* rpmb_frame_buffer;

release_mmc()
{
    if (rpmb_frame_buffer)
        free(rpmb_frame_buffer);
}

rpmb_cmd()
{
    rpmb_frame_data_sz = (req_size - sizeof(struct rpmb_req));
    if ((!rpmb_frame_buffer) || (rpmb_frame_data_sz > rpmb_frame_buffer_sz))
    {
        if (rpmb_frame_buffer)
            free(rpmb_frame_buffer);
        rpmb_frame_buffer_sz = rpmb_frame_data_sz;
        rpmb_frame_buffer = malloc(rpmb_frame_buffer_sz);
    }
    memcpy( ... )
    rpmb_route_frame( ... )
}
```